### ISSUE TITLE
KeyError: 'as_dict' [DEBUG   ] LazyLoaded nested.output

### ISSUE THREAD

**Issue by kvnaveen** [saltstack/salt#43658]
### Steps to Reproduce Issue
pip install salt
pip install azure

#### Following packages are installed and not working **salt-cloud --query**
```

adal==0.4.7
asn1crypto==0.22.0
azure==2.0.0
azure-batch==3.0.0
azure-common==1.1.8
azure-datalake-store==0.0.16
azure-graphrbac==0.30.0
azure-keyvault==0.3.6
azure-mgmt==1.0.0
azure-mgmt-authorization==0.30.0
azure-mgmt-batch==4.0.0
azure-mgmt-cdn==0.30.3
azure-mgmt-cognitiveservices==1.0.0
azure-mgmt-compute==1.0.0
azure-mgmt-containerregistry==0.2.1
azure-mgmt-datalake-analytics==0.1.6
azure-mgmt-datalake-nspkg==2.0.0
azure-mgmt-datalake-store==0.1.6
azure-mgmt-devtestlabs==2.0.0
azure-mgmt-dns==1.0.1
azure-mgmt-documentdb==0.1.3
azure-mgmt-iothub==0.2.2
azure-mgmt-keyvault==0.31.0
azure-mgmt-logic==2.1.0
azure-mgmt-monitor==0.2.1
azure-mgmt-network==1.0.0
azure-mgmt-nspkg==2.0.0
azure-mgmt-rdbms==0.1.0
azure-mgmt-redis==4.1.0
azure-mgmt-resource==1.1.0
azure-mgmt-scheduler==1.1.3
azure-mgmt-sql==0.5.3
azure-mgmt-storage==1.0.0
azure-mgmt-trafficmanager==0.30.0
azure-mgmt-web==0.32.0
azure-nspkg==2.0.0
azure-servicebus==0.21.1
azure-servicefabric==5.6.130
azure-servicemanagement-legacy==0.20.6
azure-storage==0.34.3
backports-abc==0.5
certifi==2017.7.27.1
cffi==1.11.0
chardet==3.0.4
cryptography==2.0.3
enum34==1.1.6
futures==3.1.1
idna==2.6
ipaddress==1.0.18
isodate==0.5.4
Jinja2==2.9.6
keyring==10.4.0
MarkupSafe==1.0
msgpack-python==0.4.8
msrest==0.4.14
msrestazure==0.4.14
oauthlib==2.0.4
pathlib2==2.3.0
pycparser==2.18
pycrypto==2.6.1
PyJWT==1.5.3
python-dateutil==2.6.1
PyYAML==3.12
pyzmq==16.0.2
requests==2.18.4
requests-oauthlib==0.8.0
salt==2017.7.1
scandir==1.5
singledispatch==3.4.0.3
six==1.11.0
tornado==4.5.2
urllib3==1.22
```

#### Working with following requirements 
```
adal==0.4.6
asn1crypto==0.22.0
azure==2.0.0
azure-batch==3.0.0
azure-common==1.1.8
azure-datalake-store==0.0.15
azure-graphrbac==0.30.0
azure-keyvault==0.3.5
azure-mgmt==1.0.0
azure-mgmt-authorization==0.30.0
azure-mgmt-batch==4.0.0
azure-mgmt-cdn==0.30.3
azure-mgmt-cognitiveservices==1.0.0
azure-mgmt-compute==1.0.0
azure-mgmt-containerregistry==0.2.1
azure-mgmt-datalake-analytics==0.1.6
azure-mgmt-datalake-nspkg==2.0.0
azure-mgmt-datalake-store==0.1.6
azure-mgmt-devtestlabs==2.0.0
azure-mgmt-dns==1.0.1
azure-mgmt-documentdb==0.1.3
azure-mgmt-iothub==0.2.2
azure-mgmt-keyvault==0.31.0
azure-mgmt-logic==2.1.0
azure-mgmt-monitor==0.2.1
azure-mgmt-network==1.0.0
azure-mgmt-nspkg==2.0.0
azure-mgmt-rdbms==0.1.0
azure-mgmt-redis==4.1.0
azure-mgmt-resource==1.1.0
azure-mgmt-scheduler==1.1.2
azure-mgmt-sql==0.5.3
azure-mgmt-storage==1.0.0
azure-mgmt-trafficmanager==0.30.0
azure-mgmt-web==0.32.0
azure-nspkg==2.0.0
azure-servicebus==0.21.1
azure-servicefabric==5.6.130
azure-servicemanagement-legacy==0.20.6
azure-storage==0.34.3
backports-abc==0.5
certifi==2017.7.27.1
cffi==1.10.0
chardet==3.0.4
cryptography==2.0.3
enum34==1.1.6
futures==3.1.1
idna==2.6
ipaddress==1.0.18
isodate==0.5.4
Jinja2==2.9.6
keyring==10.4.0
MarkupSafe==1.0
msgpack-python==0.4.8
msrest==0.4.11
msrestazure==0.4.11
oauthlib==2.0.2
pathlib2==2.3.0
pycparser==2.18
pycrypto==2.6.1
PyJWT==1.5.2
python-dateutil==2.6.1
PyYAML==3.12
pyzmq==16.0.2
requests==2.18.4
requests-oauthlib==0.8.0
salt==2017.7.1
scandir==1.5
singledispatch==3.4.0.3
six==1.11.0
tornado==4.5.2
urllib3==1.22

```

**Comment by gtmanfred** [comment]
Can you provide the error message and what commands we need to run to replicate this, and open a new issue?

Thanks,
Daniel

**Comment by rallytime** [comment]
Fixed via #43886
